File: streaming/src/main/java/com/welfarerobotics/welfareapplication/streaming/view/SplashActivity.java
package com.welfarerobotics.welfareapplication.streaming.view;

import android.arch.lifecycle.Observer;
import android.databinding.DataBindingUtil;
import android.support.annotation.Nullable;
import android.widget.Toast;
import com.welfarerobotics.welfareapplication.streaming.R;
import com.welfarerobotics.welfareapplication.streaming.base.BaseActivity;
import com.welfarerobotics.welfareapplication.streaming.databinding.SplashView;
import com.welfarerobotics.welfareapplication.streaming.viewmodel.SplashViewModel;

/**
 * @author : Hyunwoong
 * @when : 2019-07-03 오전 12:25
 * @homepage : https://github.com/gusdnd852
 */
public class SplashActivity extends BaseActivity<SplashView, SplashViewModel> {

    @Override protected int bindView() {
        return R.layout.splash_view;
    }

    @Override protected void observe() {
        viewModel.getSplashEvent()
                .observe(this, o -> startActivity(SignInActivity.class));
    }
}

File: streaming/src/main/java/com/welfarerobotics/welfareapplication/streaming/view/SplashActivity.kt
package com.welfarerobotics.welfareapplication.streaming.view


import android.databinding.DataBindingUtil
import android.os.Bundle
import com.welfarerobotics.welfareapplication.streaming.R
import com.welfarerobotics.welfareapplication.streaming.base.BaseActivity
import com.welfarerobotics.welfareapplication.streaming.databinding.SplashViewBinding
import com.welfarerobotics.welfareapplication.streaming.viewmodel.SplashViewModel

/**
 * @author : Hyunwoong
 * @when : 2019-07-02 오후 9:50
 * @homepage : https://github.com/gusdnd852
 */
class SplashActivity : BaseActivity() {

    lateinit var view : SplashViewBinding
    lateinit var viewModel : SplashViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        view = DataBindingUtil.setContentView(this, R.layout.splash_view)
        viewModel = factory.create(SplashViewModel::class.java)

    }
}